### PR TITLE
react-state - Logan Lee

### PR DESCRIPTION
**Pull request by loganlee-lth**
Kapture 2023-06-27 at 01 18 08

### CODE REVIEW

File: react-state/my-app/src/ToggleButton.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { useState } from 'react';
+
+export default function ToggleButton({ text, color }) {
+  const [isClicked, setIsClicked] = useState(false);
+  console.log('useState:', isClicked);
+
+  function handleClick() {
+    console.log('before setter:', isClicked);
+    setIsClicked(isClicked ? false : true);
```
Comment (shawnkost) [inline react-state/my-app/src/ToggleButton.js:9]: Works, but you can also just do `setIsClicked(!isClicked)`

File: react-state/my-app/src/ToggleButton.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    console.log('after setter:', isClicked);
+  }
+
+  return (
+    <button onClick={handleClick} style={ isClicked ? { backgroundColor: 'white' } : { backgroundColor: color } }>
```
Comment (shawnkost) [inline react-state/my-app/src/ToggleButton.js:14]: Can simplify your ternary a bit.

`style={ backgroundColor: isClicked ? 'white' : color }`

**Review by shawnkost — APPROVED** [review]
Approved, but check out my comments!

File: react-state/my-app/src/ToggleButton.js (hunk 2 above)
Comment (loganlee-lth) [reply react-state/my-app/src/ToggleButton.js]: Got it!

**Pull request merged**